app/models/pawn.rb

```diff
@@ -1,2 +1,54 @@
 class Pawn < Piece
+  def move_legal?(to_row, to_col)
+    return true if single_advance?(to_row, to_col)
+    return true if double_advance?(to_row, to_col)
+    return true if capture_move?(to_row, to_col)
+    return true if en_passant?(to_row, to_col) # need to send signal to capture logic for en passant
+    false
+  end
+
+  private
+
+  def single_advance?(to_row, to_col)
+    to_row == row + forward_one && to_col == col
+  end
+
+  def double_advance?(to_row, to_col)
+    to_row == row + forward_two && to_col == col && moves.empty?
```

Comment:
This should also verify that the piece hasn't already moved (since double advances are only valid on the first move).